### PR TITLE
checker: disallow `none` as value type in map

### PR DESCRIPTION
**Pull request by Delta456**
### <samp>🤖 Generated by Copilot at 6af324a</samp>

Prevent using `none` as a map value type in the checker module and add a test case for it. This fixes a potential compiler error or runtime crash when using invalid map literals.

### <samp>🤖 Generated by Copilot at 6af324a</samp>

* Add a check to prevent using `none` as a map value type and emit an error message ([link](https://github.com/vlang/v/pull/19582/files?diff=unified&w=0#diff-3a150553db91ada683e7bb70efe0d6d61a0c497bfdcd7c608d7f3f77282ae82cR357-R359))
* Add a test case to verify the error message for using `none` as a map value type ([link](https://github.com/vlang/v/pull/19582/files?diff=unified&w=0#diff-b216b66342f8c1dc51a2ae02eb1881c30dd2cab8c27bcf804f508e9e4fc1c6d9R1-R3), [link](https://github.com/vlang/v/pull/19582/files?diff=unified&w=0#diff-a4b9e34f5953b5de57e742139fa0b888873853b6f00088344b8e78b313319360R1))

### CODE REVIEW

File: vlib/v/checker/containers.v [hunk 1]
```diff
@@ -354,6 +354,9 @@ fn (mut c Checker) map_init(mut node ast.MapInit) ast.Type {
 			if info.value_type.has_flag(.result) {
 				c.error('cannot use Result type as map value type', node.pos)
 			}
+			if info.value_type == ast.none_type {
```
[the remark this answers is not in the document]
Comment (Delta456) [reply vlib/v/checker/containers.v:357]: What all types you think should be added?

[the remark this answers is not in the document]
Comment (JalonSolov) [reply vlib/v/checker/containers.v:357]: I don't think any types should be added.  No values should be allowed.

There seems to be a problem in understanding amongst many out there...

A map has keys that point to values, but when _declaring_ a map, you do not use values, you only use the types the values can be.

So `m := map[string]int` is valid, as it shows the _type_ of values the map can store, where `m := map[string]2` is invalid, as `2` is a value, not a type.  Same with `m := map[string]none`, since `none` is a value, not a type.

[the remark this answers is not in the document]
Comment (JalonSolov) [reply vlib/v/checker/containers.v:357]: You just have to check `!= type`, since only types are allowed.

[the remark this answers is not in the document]
Comment (medvednikov) [reply vlib/v/checker/containers.v:357]: @JalonSolov is correct here.

@Delta456 always try to make your solution more generic. Such very specific checks slow down the compiler, increase the complexity and only solve a subset of the problem.

`none` is a value, just like `2`.

We don't allow `m := map[string]2`, and yet we don't have checks like this for numbers.

[the remark this answers is not in the document]
Comment (Delta456) [reply vlib/v/checker/containers.v:357]: The checker gives an error for this.

```
file.v:1:6: error: unknown type `23`
    1 | _ := map[int]23{}
      |      ~~~~~~~~~~~~
If the code of your project is in multiple files, try with `v .` instead of `v file.v`

```

[the remark this answers is not in the document]
Comment (JalonSolov) [reply vlib/v/checker/containers.v:357]: That's the right idea, but the error message definitely needs to be improved.  Instead of `error: unknown type`, it needs to say something like `error: only recognized types allowed`.  Just need to make sure V knows that `none` is not a type.

[the remark this answers is not in the document]
Comment (Delta456) [reply vlib/v/checker/containers.v:357]: It looks like this will be a change for struct fields and all as well so I believe it would be better if we did this in another PR.

**Comment by spytheman** [comment]
This works, i.e. there is no cgen error:
```v
m := map[int]none{}
dump(m)
```
i.e. we can wait for a more generic error PR, without needing this one.
